File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        iwillstoreint=[]
        
        def recuIn(root1):
            if root1==None:
                return
            recuIn(root1.left)
            iwillstoreint.append(root1.val)
            recuIn(root1.right)
        recuIn(root)
        return iwillstoreint

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> x=new ArrayList();
        ahhhINeedToCreateOneFunction(root,x);
        return x;
    }
    public  void ahhhINeedToCreateOneFunction(TreeNode t,List<Integer> main){
        if(t!=null){
            ahhhINeedToCreateOneFunction(t.left,main);
            main.add(t.val);
            ahhhINeedToCreateOneFunction(t.right,main);
        }
    }
}